app/src/main/java/ingsw/group1/findmyphone/contacts/ContactConverterUtils.java

```diff
@@ -0,0 +1,58 @@
+package ingsw.group1.findmyphone.contacts;
+
+import androidx.room.TypeConverter;
+
+import com.eis.communication.Peer;
+
+
+/**
+ * Interface defining a converter
+ * necessary to convert {@link P} to {@link C} and vice versa.
+ * It doesn't check the validity of a contact or a peer.
+ *
+ * @author Giorgia Bortoletti
+ */
+public interface ContactConverterUtils<A extends Comparable<A>, P extends Peer<A>, C extends GenericContact<A, P>> {
```

Comment:
I understand what you are trying to do because Room converters are required to have static methods, just keep in mind that you will never be able to consider the `SMSContactCoverterUtils` class as a `ContactConverterUtils<String, SMSPeer, SMSContact>`, because the static reference to the class can't be erased.

Reply:
So how could I change this class or  `SMSContactCoverterUtils`  ?

Reply:
Just leave it like this, it would take many useless steps